src/Kernel/ConstantBlockClosure.class.st

```diff
@@ -38,11 +38,24 @@ ConstantBlockClosure class >> numArgs: numArgs literal: aLiteral [
 		literal: aLiteral
 ]
 
+{ #category : #literals }
+ConstantBlockClosure >> blockViaLiteralPath: anArray [ 
+	"return the compiledBlock for a path (see CompiledBlock>>literalPath)"
+	^ self
+]
+
 { #category : #accessing }
 ConstantBlockClosure >> literal: anObject [
 	 literal := anObject
 ]
 
+{ #category : #accessing }
+ConstantBlockClosure >> literalPath [
+	"return an array with the offsets in the literals from the homeMethod to here"
+	
+	 ^self outerCode literalPath, {self outerCode indexOfLiteral: self}
```

Comment:
looking at `CompiledBlock >> literalPath`, I'm wondering if there is a way to move the path construction responsibility to the outercode.

`CompiledBlock >> literalPathFor:` that takes a literal, so this method could simply be `^self outerCode literalPathFor: self`